tests/backend/Functions.Tests/AnalyzeTests.cs

```diff
@@ -41,14 +46,40 @@ public class AnalyzeTests
 
             WorkflowManager.Initialize();
             var workflowQueue = Substitute.For<ICollector<WorkflowQueueMessage>>();
-            var response = await Analyze.Run(request, workflowQueue, NullLogger.Instance);
+            var storage = Substitute.For<IStorage>();
+            storage.SaveToBlobAsync(Arg.Any<AnalyzeRequest>(), Arg.Any<string>()).Returns(Task.FromResult(true));
+            var response = await Analyze.Run(request, workflowQueue, storage, NullLogger.Instance);
             var body = await response.Content.ReadAsStringAsync();
 
             Assert.True(Guid.TryParse(body, out var submissionId));
 
             workflowQueue.Received().Add(Arg.Is<WorkflowQueueMessage>(x => x.SubmissionId == submissionId.ToString() && x.Stage == WorkflowStage.Analyze));
         }
 
+        [Fact]
+        public static async Task SavedRequestMatchesOriginal()
+        {
+            var gzippedAnalyzeRequestStream = typeof(AnalyzeTests).Assembly
```

Comment:
nit: logical grouping between test setup, "act" and "assert".  I think the `var response =...`

Reply:
Good suggestion! Fixed

---
In reply to: [178100247](https://github.com/Microsoft/dotnet-apiport/pull/598#discussion_r178100247) [](ancestors = 178100247)